### PR TITLE
feat: switch to halfmoon and htm/react

### PR DESCRIPTION
**Pull request by RichiCoder1**
Starts work to switch to [Halfmoon](https://www.gethalfmoon.com/) for CSS and HTM/React for JavaScript-based functionality.

Preview Pics:

## Homepage
localhost_5000_ (1)
localhost_5000_(iPad Pro) (1)

## Org Page
localhost_5000_github com_crossplane_provider-rook(iPad Pro) (1)
localhost_5000_github com_crossplane_provider-rook(iPad Pro) (3)

## CRD Page
localhost_5000_github com_crossplane_provider-azure_azure crossplane io_Provider_v1alpha3_(iPad Pro)
localhost_5000_github com_crossplane_provider-azure_azure crossplane io_Provider_v1alpha3_(iPad Pro) (1)

### CODE REVIEW

File: cmd/gitter/main.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func gitter(repos []string, r *redis.Client) error {
 
 	for _, repoURL := range repos {
 		log.Printf("Indexing repo %s...\n", repoURL)
-		dir, err := ioutil.TempDir("tmp", "doc-gitter")
+		dir, err := ioutil.TempDir(os.TempDir(), "doc-gitter")
```
Comment (RichiCoder1) [inline cmd/gitter/main.go:88]: This call was very unhappy in my env without this change. Worked in docker however.

File: cmd/doc/main.go [hunk 2]
```diff
@@ -62,11 +62,7 @@ var docTemplate = template.Must(template.New("doc.html").Funcs(
 			}
 		},
 	},
-).ParseFiles("template/doc.html", "template/analytics.html", "template/twitter.html"))
-
-var orgTemplate = template.Must(template.ParseFiles("template/org.html", "template/analytics.html", "template/twitter.html"))
-var newTemplate = template.Must(template.ParseFiles("template/new.html", "template/analytics.html", "template/twitter.html"))
-var homeTemplate = template.Must(template.ParseFiles("template/home.html", "template/analytics.html", "template/twitter.html"))
+).ParseGlob("template/*.html"))
```
Comment (RichiCoder1) [inline cmd/doc/main.go:65]: Thoughts on using something like https://github.com/unrolled/render to get layouts and de-dupe things a bit more?

File: cmd/gitter/main.go (hunk 1 above)
Comment (hasheddan) [reply cmd/gitter/main.go:88]: Yeah there is some funky logic here I need to correct, I'll open a PR and tag you on it :+1:

File: cmd/doc/main.go (hunk 2 above)
Comment (hasheddan) [reply cmd/doc/main.go:65]: Not super familiar, but I am all for it if it simplifies some of the gross logic that currently exists in the template :sweat_smile:

Comment (RichiCoder1) [reply cmd/doc/main.go:65]: Noted! I'll give it a try

**Comment by RichiCoder1** [comment]
Still some parity and cleanup work to be done, but here's the new org and doc screens:

localhost_5000_github com_crossplane_provider-rook_database rook crossplane io_CockroachCluster_v1alpha1_(iPad Pro)
localhost_5000_github com_crossplane_provider-rook(iPad Pro)

**Comment by RichiCoder1** [comment]
@hasheddan LMK what you think! A lot more I could do, but figured this is a good stopping point for the conversion. See the description for pics, and since tis is all cdn you can also clone & run locally.

Side note: If this site gets any more complicated, you might wanna go the spa route haha.

File: cmd/doc/main.go [hunk 3]
```diff
@@ -53,23 +57,35 @@ type SchemaPlusParent struct {
...
-		"plusParent": func(p string, s map[string]apiextensions.JSONSchemaProps) *SchemaPlusParent {
-			return &SchemaPlusParent{
-				Parent: p,
-				Schema: s,
-			}
+var page = render.New(render.Options{
+	Extensions:    []string{".html"},
+	Directory:     "template",
+	Layout:        "layout",
+	IsDevelopment: os.Getenv(envAnalytics) == "true",
```
Comment (hasheddan) [inline cmd/doc/main.go:64]: should this be checking the new `envDevelopment` instead?

File: cmd/doc/main.go [hunk 4]
```diff
@@ -118,6 +130,18 @@ func main() {
 	start()
 }
 
+func getPageData(r *http.Request, disableNavBar bool) pageData {
+	var isDarkMode = false
+	if cookie, err := r.Cookie(cookieDarkMode); err == nil && cookie.Value == "dark-mode" {
```
Comment (hasheddan) [inline cmd/doc/main.go:135]: ah nice, so it will remember a user's preference? 🎉

**Review by hasheddan — COMMENTED** [review]
@RichiCoder1 I need to do a little more poking around locally, but this is looking great! Thank you so much for your work on this! Was there any additional updates you were planning for this PR or do you feel pretty good about where it's at currently?

File: cmd/doc/main.go (hunk 3 above)
Comment (RichiCoder1) [reply cmd/doc/main.go:64]: Derp. Fixed

File: cmd/doc/main.go (hunk 4 above)
Comment (RichiCoder1) [reply cmd/doc/main.go:135]: Exactly! It's actually possible for this to work without a server side but, but that was causing a brief flash so I figured I'd add it here.

**Comment by RichiCoder1** [comment]
> Was there any additional updates you were planning for this PR or do you feel pretty good about where it's at currently?

I think for this PR, I'm feeling pretty good about it.

**Comment by RichiCoder1** [comment]
One caveat worth not: To make this build-tool-less, I'm using a lot of modern features. So will only support evergreen (eg, auto-updating) browsers.

**Comment by hasheddan** [comment]
@RichiCoder1 this is looking awesome! Thanks for your work here :) Apologies for the delay on reviews, hoping to get it merged and live ASAP!

**Comment by RichiCoder1** [comment]
All good! Take your time, it's a big change haha. Am excited to get it out and possibly iterate though. I think there's still some possibly cool ways to improve the UX

**Comment by hasheddan** [comment]
@RichiCoder1 thanks again for your patience here! I have tested this out locally and found one minor fix that needs to be made, but otherwise this looks awesome! I am going to go ahead and merge, open a PR for the fix and then let's get this out and get some feedback :) I totally agree that it might make sense to move towards a SPA as this continues to grow in complexity. Let me know if you want to set up a time to chat about that :+1:

**Pull request merged**